chain/ethereum/src/network_indexer/convert.rs

```diff
@@ -113,9 +169,101 @@ impl TryIntoEntity for &BlockWithOmmers {
                     .collect::<Vec<_>>()
                     .into(),
             ),
+            ("ommerHash", inner.uncles_hash.into()),
+            (
+                "transactions",
+                (inner
+                    .transactions
+                    .iter()
+                    .map(move |transaction| transaction.hash.to_entity_id())
+                    .collect::<Vec<_>>()
+                    .into()),
+            ),
+            ("logCount", (logs.len() as i32).into()),
+            ("logs", logs.into()),
             ("size", inner.size.into()),
             ("sealFields", inner.seal_fields.clone().into()),
-            ("isOmmer", false.into()),
+        ] as Vec<(_, Value)>))
+    }
+}
+
+impl TryIntoEntity for &LoadedTransaction {
+    fn try_into_entity(self) -> Result<Entity, Error> {
+        let transaction = &self.transaction;
+        let receipt = &self.receipt;
+
+        Ok(Entity::from(vec![
+            ("id", transaction.hash.to_entity_id().into()),
+            ("hash", transaction.hash.into()),
+            ("nonce", transaction.nonce.into()),
+            ("index", transaction.transaction_index.unwrap().into()),
+            ("from", transaction.from.into()),
+            ("to", transaction.to.map_or(Value::Null, |to| to.into())),
+            ("value", transaction.value.into()),
+            ("gasPrice", transaction.gas_price.into()),
+            ("gas", transaction.gas.into()),
+            (
+                "block",
+                format!("{:x}", transaction.block_hash.unwrap()).into(),
+            ),
+            ("blockNumber", transaction.block_number.unwrap().into()),
+            ("logCount", (receipt.logs.len() as i32).into()),
+            (
+                "logs",
+                receipt
+                    .logs
+                    .iter()
+                    .cloned()
+                    .map(|log| Log::from(log).to_entity_id())
+                    .collect::<Vec<_>>()
+                    .into(),
+            ),
+            ("status", receipt.status.into()),
+            ("gasUsed", receipt.gas_used.into()),
+            ("cumulativeGasUsed", receipt.cumulative_gas_used.into()),
+            (
+                "createdContract",
+                receipt
+                    .contract_address
+                    .map_or(Value::Null, |address| address.into()),
+            ),
+        ] as Vec<(_, Value)>))
+    }
+}
+
+impl TryIntoEntity for &Log {
+    fn try_into_entity(self) -> Result<Entity, Error> {
+        let inner = &self.0;
+
+        Ok(Entity::from(vec![
+            (
+                "id",
+                format!(
+                    "{:x}-{}",
+                    inner.block_hash.unwrap(),
+                    inner.log_index.unwrap()
+                )
```

Comment:
I use "{block_hash}-{log_index}" here to build unique ids for log entities,  another option is to use "{transaction_hash}-{log_index}".  It's not obvious to me which approach will be more clear to users, or will support more useful queries.
   
I don't foresee either id structure being necessary for querying since one can query for transaction or block entities at the top level and get logs as a sub-entity. For example, getting all logs from a specific block or transaction is easily doable without filtering on the prefix of the log id.

Reply:
Yeah, I think the IDs should be treated as opaque values. Nobody should really filter logs with `id_starts_with: { "sometxhash-" }`.

What we should really do though is use `Bytes` for all these IDs because they are more efficiently stored and indexed by Postgres.